WordPressComStatsiOS/WordPressComStatsiOS/UI/WPStatsGraphViewController.m

```diff
@@ -208,6 +217,44 @@ - (void)setVisits:(StatsVisits *)visits forSummaryType:(StatsSummaryType)summary
 
 #pragma mark - Private methods
 
+- (CGFloat)calculateDateFontSize
+{
+    // step through each of the dates
+    // and calculate what font size each would actually be displayed at
+    // then return the smallest font size out of all of them
+    CGFloat smallestFontSize = [[WPStyleGuide axisLabelFont] pointSize];
+
+    for (StatsSummary *summary in self.statsData) {
+        NSInteger index = (NSInteger)[self.statsData indexOfObject:summary];
+        NSIndexPath *indexPath = [NSIndexPath indexPathForItem:index inSection:0];
+
+        WPStatsGraphBarCell *cell = [self collectionView:self.collectionView cellForItemAtIndexPath:indexPath];
+        CGFloat currentFontSize = [self fontSizeForString:cell.barName inRect:cell.frame];
+
+        if (index == 0) {
+            smallestFontSize = currentFontSize;
+        } else if (currentFontSize < smallestFontSize) {
+            smallestFontSize = currentFontSize;
+        }
+    }
+
+    return smallestFontSize;
+}
+
+- (CGFloat)fontSizeForString: (NSString *)string inRect: (CGRect)rect
+{
+    UIFont *font = [WPStyleGuide axisLabelFont];
```

Comment:
Just a thought! no need to change anything:

By making the UIFont a parameter, you'd be able to encapsulate this into a `NSString+FontSize` extension, which could be reused anywhere. The method signature could look like this:

```
NSString *something = @"Aloha!";
CGFloat calculatedSize = [something sizeWithFont:(UIFont *)font containerSize:(CGSize)size];
```